MaxText/max_utils.py

```diff
@@ -351,3 +347,18 @@ def _cross_entropy_with_logits_bwd(
 
 cross_entropy_with_logits.defvjp(_cross_entropy_with_logits_fwd,
                                  _cross_entropy_with_logits_bwd)
+
+
+def get_abstract_state(model, tx, config, rng, mesh):
```

Comment:
maxtext_util? Seems maxtext specific!

Reply:
Indeed, refactored max_utils and maxtext_utils, I don't think a follow up PR is needed anymore for the split

Reply:
Original thinking was get_abstract_state was a tightly bound to setup_initial_state so was gonna wait for refactor PR to move both together. However now I have get_abstract_state in maxtext_utils and will wait to move other methods for refactor PR. This PR creates no new methods in max_utils.py since all happen to maxtext specific.